scout/adapter/mongo/variant.py

```diff
@@ -882,16 +883,19 @@ def case_variants_count(self, case_id, institute_id, update_case=True):
                 }
         """
         LOG.info(
-            "Retrieving variants by categori for case: {0}, institute: {1}".format(
+            "Retrieving variants by category for case: {0}, institute: {1}".format(
                 case_id, institute_id
             )
         )
-        # if case has stats and no update is needed, return variant count
+
         case_obj = self.case(case_id=case_id)
-        if case_obj.get("variants_stats") and update_case is False:
+        variants_stats = case_obj.get("variants_stats") or {}
```

Comment:
Why not use a default for get? Were there empty lists?

Reply:
Cases don't have a variant_stats key when they are first loaded. It gets created when you first visit the case variantS page. Does this answer your wuestion?

Reply:
Not quite: I was wondering why not
```suggestion
        variants_stats = case_obj.get("variants_stats",{})
```

Reply:
I think it works, so no worries, just curious!

Reply:
Yees, but if they dont exist, the usual get("variant_stats", {}) would work. The only difference I can think of would be that variant_stats exists, and gives some other non-true value than None or `{}`, like `[]`?

Reply:
No there shouldn't be any other values than a populated dictionary. get("variant_stats", {}) would work

Reply:
Doesn't look like this change works (see error message below). Reproduce it by running `scout setup demo` and then `scout --demo load research -c internal_id --force`. Could it be that somewhere along the commands and before calling this function a case.variant_stats is created with a None value?

in that case None would be picked instead of {}, causing the error. I'll revert to the previous code that looks safer